### ISSUE TITLE
Non-uniform behavior with Qt::AlignmentFlag

### ISSUE THREAD

**Issue by albertosottile** [mottosso/Qt.py#252]
When working with PySide and PySide2, the following code snippets work perfectly:
```
layout = QtWidgets.QVBoxLayout()
layout.setAlignment(Qt.AlignTop & Qt.AlignLeft)
```

or
```
layout = QtWidgets.QVBoxLayout()
layout.addWidget(widget, 1, 0).
```

When working with PyQt5, both these two snippets stop working. The first one gives this error:
```
TypeError: arguments did not match any overloaded call:
  setAlignment(self, QWidget, Union[Qt.Alignment, Qt.AlignmentFlag]): argument 1
 has unexpected type 'int'
  setAlignment(self, QLayout, Union[Qt.Alignment, Qt.AlignmentFlag]): argument 1
 has unexpected type 'int'
  setAlignment(self, Union[Qt.Alignment, Qt.AlignmentFlag]): argument 1 has 
unexpected type 'int'
```

while the second causes this one: 
```
TypeError: addWidget(self, QWidget, stretch: int = 0, alignment: Union[
Qt.Alignment, Qt.AlignmentFlag] = 0): argument 3 has unexpected type 'int'
```

PyQt5 for some reason stores these values as int but requires them to be of Qt::Alignment type. PySide and PySide2 store these value as Qt::Alignment but also accept int as input.

**Comment by mottosso** [comment]
Thanks for reporting this @alby128. I think this is what you're looking for.

```python
layout.setAlignment(Qt.AlignTop | Qt.AlignLeft)
```

Notice the `|` in place of `&`. Flags in Qt are typically OR together, but I'm not sure about this one in particular.

**Comment by alby128** [comment]
I thought about that, and indeed the first snippet works using the | operator. Yet I do not understand why the & works with PySide and PySide2, even when they are used without Qt.py. 

Any idea about the `addWidget` method? Thanks

**Comment by mottosso** [comment]
> Yet I do not understand why the & works with PySide and PySide2

I think because the `|` operator produces an object of `Qt.Alignment`, whereas `&` produces an integer. And PySide just so happens to support both. 

```python
>>> QtCore.Qt.AlignTop & QtCore.Qt.AlignLeft
0
>>> QtCore.Qt.AlignTop | QtCore.Qt.AlignLeft
<PyQt5.QtCore.Alignment object at 0x0000000002F690B8>
>>>
```

Are you sure it produces the expected result though? I'd expect the result of `&` to be jibberish to Qt.

> Any idea about the addWidget method? Thanks

Is it an option to use only the Alignment objects? This works for me in PyQt4 and PySide.

```python
>>> from Qt import QtWidgets, QtCore
>>> import sys
>>> app = QtWidgets.QApplication(sys.argv)
>>> layout = QtWidgets.QVBoxLayout()
>>> widget = QtWidgets.QPushButton()
>>> layout.addWidget(widget, 1, QtCore.Qt.AlignLeft)
>>> layout.addWidget(widget, 1, QtCore.Qt.AlignLeft | QtCore.Qt.AlignRight)
```

**Comment by alby128** [comment]
I noticed that too, but for some reason the result of & in PySide and PySide2 is not an int.
With PySide2 this is the result:
```
>>> from PySide2.QtCore import Qt
>>> Qt.AlignTop & Qt.AlignLeft
<Alignment object at 0x108abe0f0>
>>> Qt.AlignTop | Qt.AlignLeft
<Alignment object at 0x108abe108>
```
and with PySide:
```
>>> from PySide.QtCore import Qt
>>> Qt.AlignTop & Qt.AlignLeft
<Alignment object at 0x1027db0f0>
>>> Qt.AlignTop | Qt.AlignLeft
<Alignment object at 0x1027db108>
```

About the `addWidget`, that was the temporary solution I used to test the code so far, although I do not remember the right flag for 0.

All the solution you are proposing work fine, but they require extensive changes in the codebase that are not necessary with PySide and PySide2. Wouldn't it be better if Qt.py handles this, instead?

**Comment by mottosso** [comment]
> but for some reason the result of & in PySide and PySide2 is not an int.

Yes, but keep an eye out for whether the result of `|` and `&` are actually the same. I edited the comment above to include an example of how this might not be the case.

```python
>>> 1 & 2
0
>>> 1 | 2
3
>>>
```

> Wouldn't it be better if Qt.py handles this, instead?

If by handle you mean modify `AlignLeft`  and friends than that could prove difficult. Qt.py only maps things from one binding to another. Best it could do is give you a `QtCompat.AlignLeft` that you could use in place of the original ones.

Or did you mean that your codebase currently uses integers? In that case, you'd be out of luck as it would mean modifying the instanced class of `QLayout` and friends, overloading their call to `setAlignment` and `addWidget`.

**Comment by alby128** [comment]
I am trying to actually read the result of these operations, so far with no luck. In any case (Qt.AlignLeft & Qt.AlignTop) == (Qt.AlignLeft | Qt.AlignTop) gives False, so I think you are right. 

Our codebase currently uses integers and &, and it works with Qt.py on PySide and PySide2. I was attempting to extend its compatibility to PyQt5, but I did not want to change all these calls.

**Comment by mottosso** [comment]
Ok, if you manage to find a way of implementing it that works across bindings, but doesn't introduce side-effects for vanilla use of the binding, then I'd welcome a PR for it.
